SonarQube.Common/ConsoleLogger.cs

```diff
@@ -25,49 +27,114 @@ namespace SonarQube.Common
     /// </summary>
     public class ConsoleLogger : ILogger
     {
+        private enum MessageType
+        {
+            Info,
+            Debug,
+            Warning,
+            Error
+        }
+
+        private class Message
+        {
+            private readonly MessageType messageType;
+            private readonly string finalMessage;
+
+            public Message(MessageType messageType, string finalMessage)
+            {
+                this.messageType = messageType;
+                this.finalMessage = finalMessage;
+            }
+
+            public MessageType MessageType {  get { return this.messageType; } }
```

Comment:
you could use this instead and remove the private members:
    public MessageType MessageType { get; }
    public string FinalMessage { get; }

this way the properties are settable only in constructors.